### DISCUSSION TITLE
What is the best way to get the job failure errstr without using `wait`?

### DISCUSSION THREAD

**Discussion by SteVwonder** [flux-framework/flux-core#3496] [Q&A] [answered]
`flux.job.wait` returns a jobid, failure boolen, and errstr if there was a failure.  If you aren't using the wait interface (like in the new `FluxEventExecutor` class), then what is the best way to get the errstr?  I couldn't find it in the main eventlog, guest.exec eventlog, or the KVS.

CC @jameshcorbett

**Comment by grondo** [comment, marked as answer]
The `errstr` is not stored anywhere, but is created by the job manager as it responds to a wait request when the job has exited.

https://github.com/flux-framework/flux-core/blob/master/src/modules/job-manager/wait.c#L72

The string itself is informational and is not meant to be part of job provenance, since it can be recreated from the `finish` event status.

A similar message (probably not exactly the same) is returned by `flux job status` by processing the eventlog, e.g:
```
$ flux job status -v ƒczpiyJKR
flux-job: ƒczpiyJKR: exited with exit code 1
```

**Reply by SteVwonder** [reply]
> A similar message (probably not exactly the same) is returned by flux job status by processing the eventlog

Ah! Good to know. I thought the errstr contained additional information beyond the exit code.  In that case, I think exit code is sufficient.  Thanks @grondo!

**Reply by grondo** [reply]
Ah, well, technically the `errstr` will  also describe job exceptions, so I suppose it isn't *only* the exit code, but all the information is indeed in the eventlog. I wonder if we should have a convenience function that produces the `errstr` as in `flux job status` or the job manager's `decode_job_result()`
